slider/src/main/java/io/monstarlab/mosaic/slider/Slider.kt

```diff
@@ -0,0 +1,149 @@
+import androidx.compose.foundation.focusable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.LayoutDirection
+
+/**
+ * A composable function that creates a slider UI component.
+ *
+ * @param value the current value of the slider
+ * @param onValueChange a callback function invoked when the slider value changes
+ * @param enabled - determines whether the user can interact with the slide or not
+ * @param colors the colors used to customize the appearance of the slider
+ * @param modifier the modifier to be applied to the slider
+ * @param valueDistribution the strategy for distributing slider values
+ * @param range the range of values the slider can represent
+ * @param interactionSource the interaction source used to handle user input interactions
+ * @param thumb the composable function used to render the slider thumb
+ */
+@Composable
+public fun Slider(
+    value: Float,
+    onValueChange: (Float) -> Unit,
+    colors: SliderColors,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    valueDistribution: SliderValueDistribution = SliderValueDistribution.Linear,
+    range: ClosedFloatingPointRange<Float> = 0f..1f,
+    disabledRange: ClosedFloatingPointRange<Float> = EmptyRange,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    thumb: @Composable (
+        SliderState,
+    ) -> Unit = { DefaultSliderThumb(colors = colors, enabled = enabled) },
+) {
+    val state = rememberSliderState(value, range, valueDistribution, disabledRange)
+    state.onValueChange = onValueChange
+    state.value = value
+
+    // Workaround for the initial value that might belong into disabled range
+    // The initial value is remembered
+    val initialValue = remember { mutableFloatStateOf(value) }
+
+    // In case initial value differs from the value inside the state
+    // this means the value has been coerced into it
+    // for this case the change of value must be explicitly notified to the receiver
+    LaunchedEffect(initialValue) {
+        if (initialValue.value != state.value) {
+            onValueChange(state.value)
+        }
+    }
+
+    Slider(
+        state = state,
+        interactionSource = interactionSource,
+        modifier = modifier,
+        thumb = thumb,
+        colors = colors,
+        enabled = enabled,
+    )
+}
+
+/**
+ * A composable function that creates a slider UI component.
```

Comment:
Missing empty line after:  "A composable function that creates a slider UI component." to ensure consistent documentation for both sliders.

Reply:
Fixed